### PR TITLE
Fqsns

### PR DESCRIPTION
**Pull request by goodboy**
`fqsn` = "fully qualified symbol name"

Basically the idea here is for all client-ish code (aka programs/actors that ask the provider agnostic layers in the stack for data) should be able to tell which backend / venue / derivative each data feed/flow is from by an explicit string key of the current form `<instrumentname>.<venue>.<suffixwithmetadata>.<brokerbackendname>`.

I have thoughts that we should actually change this to be more like an "attr lookup" (like how the web should have done urls, but marketting peeps ruined it etc. etc.):

`<broker>.<venue>.<instrumentname>.<suffixwithmetadata>`

This preps for new derivatives search support in the `ib` backend coming in #294 (which will get rebased and likely renamed after this).

----
Moar,

- there's some fixes to the data (feed) layer to be more tolerant to IPC failures
- there might be some dependency custom fork changes i have to include, gotta dig through what's missing - CI and all..

### CODE REVIEW

**Comment by goodboy** [comment]
Bleh, CI is actually failing due to this change:

```
requirements.txt (line 4))
  Cloning git://github.com/goodboy/tractor.git (to revision master) to ./src/tractor
  Running command git clone --filter=blob:none --quiet git://github.com/goodboy/tractor.git /home/runner/work/piker/piker/src/tractor
  fatal: remote error:
    The unauthenticated git protocol on port 9418 is no longer supported.
  Please see https://github.blog/2021-09-01-improving-git-protocol-security-github/ for more information.
```

So i guess even more hassle to test things 😂

File: piker/brokers/kraken.py [hunk 1]
```diff
@@ -18,9 +18,9 @@
 Kraken backend.
 
 """
-from contextlib import asynccontextmanager
+from contextlib import asynccontextmanager as acm
```
Comment (iamzoltan) [inline piker/brokers/kraken.py:21]: this is definitely cleaner

File: piker/clearing/_ems.py [hunk 2]
```diff
@@ -20,7 +20,7 @@
 """
 from contextlib import asynccontextmanager
 from dataclasses import dataclass, field
-from math import isnan
+# from math import isnan
```
Comment (iamzoltan) [inline piker/clearing/_ems.py:23]: does this still need to be here?

File: piker/clearing/_paper_engine.py [hunk 3]
```diff
@@ -32,6 +32,7 @@
 
 from .. import data
 from ..data._normalize import iterticks
+from ..data._source import uncons_fqsn
```
Comment (iamzoltan) [inline piker/clearing/_paper_engine.py:35]: uncons should likely be changed to unpack

File: piker/data/_source.py [hunk 4]
```diff
@@ -91,31 +92,113 @@ def ohlc_zeros(length: int) -> np.ndarray:
     return np.zeros(length, dtype=base_ohlc_dtype)
 
 
+def uncons_fqsn(fqsn: str) -> tuple[str, str, str]:
```
Comment (iamzoltan) [inline piker/data/_source.py:95]: change to unpack_fqsn

File: piker/clearing/_client.py [hunk 5]
```diff
@@ -194,21 +196,22 @@ async def open_ems(
...
 
         async with (
-
             # connect to emsd
             portal.open_context(
 
                 _emsd_main,
-                broker=broker,
-                symbol=symbol.key,
+                fqsn=fqsn,
```
Comment (iamzoltan) [inline piker/clearing/_client.py:214]: some docs somewhere to explain what `fqsn` stands for would be welcomed

**Review by iamzoltan — COMMENTED** [review]
looks good to me. mainly some docs on the new nomenclature as well as a func name change for clarity.

I dig it.

File: piker/clearing/_ems.py (hunk 2 above)
Comment (goodboy) [reply piker/clearing/_ems.py:23]: i don't think so, i'll scan the old comments to make sure.

File: piker/clearing/_paper_engine.py (hunk 3 above)
Comment (goodboy) [reply piker/clearing/_paper_engine.py:35]: right, good call.

File: piker/clearing/_client.py (hunk 5 above)
Comment (goodboy) [reply piker/clearing/_client.py:214]: yah no kidding 😂 

the description in this PR would probably me more or less sufficient yah?

Comment (iamzoltan) [reply piker/clearing/_client.py:214]: definitely, but it would be nice to have it in the code somewhere

Comment (goodboy) [reply piker/clearing/_client.py:214]: added as doc string on `Symbol.front_fqsn()`.

File: piker/data/_source.py [hunk 6]
```diff
@@ -131,37 +214,44 @@ def front_feed(self) -> tuple[str, str]:
...
+    def front_fqsn(self) -> str:
+        '''
+        fqsn = "fully qualified symbol name"
 
-@validate_arguments
-def mk_symbol(
+        Basically the idea here is for all client-ish code (aka programs/actors
+        that ask the provider agnostic layers in the stack for data) should be
+        able to tell which backend / venue / derivative each data feed/flow is
+        from by an explicit string key of the current form:
```
Comment (iamzoltan) [inline piker/data/_source.py:231]: hot hot hot

**Review by iamzoltan — APPROVED** [review]
looks gouda to me my dude

**Review by guilledk — APPROVED** [review]

**Pull request merged**